src/options.cc

```diff
@@ -67,6 +70,29 @@ bool Options::addDefaultHandler(std::shared_ptr<NginxConfigStatement> handler_co
     return addHandler(type, path, *handler_config->child_block_);
 }
 
+bool Options::addAuthentication(std::shared_ptr<NginxConfigStatement> auth_config) {
+    if (auth_config->tokens_.size() != REALM_SIZE) {
+        std::cerr << "A path has not been specified for the realm.\n";
+        return false;
+    }
+
+    auth = new Authentication();
+    std::map<std::string, std::string> AuthMap;
+    for (auto& statements : auth_config->child_block_->statements_) {
+        if (statements->tokens_.size() != REALM_CRED_SIZE) {
+            std::cerr << "Not a valid credential statement. Need " << 
+                REALM_CRED_SIZE << " tokens.\n";
+            return false;
+        }
+        if(statements->tokens_[0] == "credential") {
+            std::string username = statements->tokens_[1];
+            std::string pass = statements->tokens_[2];
+            AuthMap[username] = pass;
+        }
+    }
+    auth->addRealm(new AuthenticationRealm(auth_config->tokens_[VAL], AuthMap));
+    return true;
```

Comment:
Ive set auth in options loadfromstream function before the stream read of the config file.